sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -48,6 +48,34 @@ private[sql] object ParquetFilters {
     override def inverseCanDrop(statistics: Statistics[T]): Boolean = false
   }
 
+  object StringFilter extends Enumeration {
+    type Mode = Value
+    val STARTS_WITH, ENDS_WITH, CONTAINS = Value
+  }
+
+  case class StringFilter(
+    v: java.lang.String,
+    mode: StringFilter.Mode) extends UserDefinedPredicate[Binary] with Serializable {
+
+    private val compare = mode match {
+      case StringFilter.STARTS_WITH =>
+        (x: java.lang.String) => x.startsWith(v)
+      case StringFilter.ENDS_WITH =>
+        (x: java.lang.String) => x.endsWith(v)
+      case StringFilter.CONTAINS =>
+        (x: java.lang.String) => x.contains(v)
+    }
+
+    override def keep(value: Binary): Boolean = {
+      val str = value.toStringUsingUTF8()
+      compare(str)
+    }
```

Comment:
Another thing that concerns me is performance. The `keep` method is called over *every* single string value, while `Binary.toStringUsingUTF8` can be pretty slow. I think we can apply the same trick used in `CatalystStringConverter` and use a mutable `UTF8String` here to do all the string operations.